bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/launch/LaunchUtils.java

```diff
@@ -145,9 +166,60 @@ public static void initConfigurableVariables(Path configFilePath, Map<Module, Va
         }
     }
 
-    public static Path getConfigPath() {
-        Map<String, String> envVariables = System.getenv();
-        return Paths.get(envVariables.getOrDefault(CONFIG_ENV_VARIABLE, Paths.get(RuntimeUtils.USER_DIR,
-                                                                                  CONFIG_FILE_NAME).toString()));
+    public static TomlDetails getConfigurationDetails() {
+        List<Path> paths = new ArrayList<>();
+        Map<String, String> envVars = System.getenv();
+        String secret = populateSecretConfigDetails(paths, envVars);
+        String configContent = populateConfigDetails(paths, envVars);
+        return new TomlDetails(paths.toArray(new Path[0]), secret, configContent);
+    }
+
+    private static String populateConfigDetails(List<Path> paths, Map<String, String> envVars) {
+        String configContent = null;
+        if (envVars.containsKey(CONFIG_FILES_ENV_VARIABLE)) {
+            String[] configPathList = envVars.get(CONFIG_FILES_ENV_VARIABLE).split(File.pathSeparator);
+            for (String pathString: configPathList) {
+                Path path = Paths.get(pathString);
+                if (!Files.exists(path)) {
```

Comment:
The `Files.exists()` checks are moved out of providers to avoid throwing errors for secret files.